File: python/src/primes.py
import math

def is_prime(n: int) -> bool:
    limit = math.isqrt(n)
    return all(n % d != 0 for d in range(2, limit + 1))

primes: list[int] = [n for n in range(2, 101) if is_prime(n)]
print(primes)

File: rust/src/bin/primes.rs
fn is_prime(n: i32) -> bool {
    let limit = (n as f64).sqrt() as i32;
    (2..=limit).all(|d| n % d != 0)
}

fn main() {
    let primes: Vec<i32> = (2..=100).filter(|&n| is_prime(n)).collect();
    println!("{:?}", primes);
}
